compose/compose.go

```diff
@@ -28,7 +31,9 @@ func SquashErrors(errs []error) error {
 	var s []string
 
 	for _, err := range errs {
-		s = append(s, err.Error())
+		if err != nil {
+			s = append(s, err.Error())
+		}
 	}
 
 	return fmt.Errorf("%s", strings.Join(s, "; "))
```

Comment:
If s is an empty list (which can happen if there are only nil errors) then we should return nil here.